source/mir/bignum/low_level_view.d

```diff
@@ -621,28 +630,100 @@ struct BigUIntView(W, WordEndian endian = TargetEndian)
+                        continue;
+                }
                 default: return false;
             }
+            ++j;
+            static if (allowUnderscores) recentUnderscore = false;
+            // how far do we need to shift to get to the top 4 bits
             enum s = W.sizeof * 8 - 4;
+            // shift number to the top most 4 bits
             W cc = cast(W)(W(c) << s);
+            // shift unsigned right 4 bits
             current >>>= 4;
+            // add number to top most 4 bits of current var
             current |= cc;
-            if (i % (W.sizeof * 2) == 0)
+            if (j % (W.sizeof * 2) == 0) // is this packed var full? 
             {
-                rdata.front = current;
-                rdata.popFront;
+                work.mostSignificant = current;
                 current = 0;
+                if (_expect(work.coefficients.length < coefficients.length, true))
+                {
+                    work = topLeastSignificantPart(work.coefficients.length + 1);
+                }
+                else if (i < str.length) // if we've run out of coefficients before reaching the end of the string, error
+                {
+                    return false;
+                }
             }
         }
         while(i < str.length);
+
+        static if (allowUnderscores) 
+        {
+            // check for a underscore at the beginning or the end
+            if (recentUnderscore || str[$ - 1] == '_') return false;
+        }
+
         if (current)
         {
-            current >>>= 4 * (W.sizeof * 2 - i % (W.sizeof * 2));
-            rdata.front = current;
+            current >>>= 4 * (W.sizeof * 2 - j % (W.sizeof * 2));
+            work.mostSignificant = current;
         }
+
+        coefficients = coefficients[0 .. (j / (W.sizeof * 2) + (j % (W.sizeof * 2) != 0))];
+
         return true;
     }
 
+    static if (W.sizeof == size_t.sizeof && endian == TargetEndian)
+    ///
+    version(mir_bignum_test)
+    @safe pure
+    unittest
+    {
+        auto view = BigUIntView!size_t.fromHexString!(char, true)("abcd_efab_cdef");
+        assert(cast(ulong)view == 0xabcd_efab_cdef);
+    }
+
+    static if (W.sizeof == size_t.sizeof && endian == TargetEndian)
+    ///
+    version(mir_bignum_test)
+    @safe pure
+    unittest
+    {
+        bool caught = false;
+        try {
+            auto view = BigUIntView!size_t.fromHexString!(char, true)("abcd_efab_cef_");
+        } catch (Exception e) {
+            caught = true;
+        }
+
+        assert(caught);
+    }
+
+    static if (W.sizeof == size_t.sizeof && endian == TargetEndian)
+    ///
+    version(mir_bignum_test)
+    @safe pure
+    unittest
+    {
+        bool caught = false;
+        try {
+            auto view = BigUIntView!size_t.fromHexString!(char, true)("abcd__efab__cef");
+        } catch (Exception e) {
+            caught = true;
+        }
+
+        assert(caught);
+    }
```

Comment:
please unite all failure test into single one.  Needs test that check number like "_abc".